config/management/src/smoke_test.rs

```diff
@@ -202,38 +190,23 @@ fn check_connectivity(swarm: &mut LibraSwarm, expected_peers: i64) -> bool {
     true
 }
 
-fn attach_validator_full_node(
-    helper: &StorageHelper,
-    validator_config: &mut NodeConfig,
-    swarm_path: &Path,
-    ns: String,
-) -> NodeConfig {
+fn attach_validator_full_node(validator_config: &mut NodeConfig) -> NodeConfig {
     // Create two vfns, we'll pass one to the validator later
-    let fn_vfn = NetworkConfig::network_with_id(NetworkId::vfn_network());
-    let v_vfn = NetworkConfig::network_with_id(NetworkId::vfn_network());
-
-    let mut full_node_config = NodeConfig::default();
-    full_node_config.full_node_networks = vec![fn_vfn, v_vfn];
+    let mut full_node_config = NodeConfig::default_for_validator_full_node();
     full_node_config.randomize_ports();
 
-    // Now let's swap the full node network on the validator with the PFN on the fullnode, since
-    // that's the VFN and what we want clients to connect to.
-    let internal_fn = full_node_config.full_node_networks.swap_remove(1);
-    let pfn = validator_config.full_node_networks.swap_remove(0);
-    full_node_config.full_node_networks.push(pfn);
-    validator_config.full_node_networks.push(internal_fn);
+    // The FN's first network is the external, public network, it needs to swap listen addresses
+    // with the validator's VFN and to copy it's key access:
+    let pfn = &mut full_node_config.full_node_networks[0];
+    let v_vfn = &mut validator_config.full_node_networks[0];
+    pfn.identity = v_vfn.identity.clone();
+    let temp_listen = v_vfn.listen_address.clone();
+    v_vfn.listen_address = pfn.listen_address.clone();
+    pfn.listen_address = temp_listen;
```

Comment:
Is it possible to just set them correctly in the first place rather than swap them?

Reply:
we can probably move toward that direction, but we'd have to change the current genesis structure and start a full node with each libra-node... maybe a future PR?